client/client_test.go

```diff
@@ -0,0 +1,256 @@
+package client
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strconv"
+	"syscall"
+	"testing"
+	"time"
+)
+
+// The IP adress of the test server
+const TestServerIP = "127.0.0.1"
+
+// The port that the server is listening on
+const TestServerPort = 8022
+
+// startTestServer starts a server listening at `TestServerIP` on
+// `TestServerPort`.  Need to later manually tear down the server.
+func startTestServer() (int, string, error) {
+	dir, err := ioutil.TempDir("", "TestServer")
+	if err != nil {
+		return 0, "", err
+	}
+
+	cmd := exec.Command("go",
```

Comment:
So I should refactor this into search-server and write new tests with fake servers here?